src/core/Akka.TestKit/Internal/AsyncQueue.cs

```diff
@@ -86,15 +91,14 @@ public bool TryTake(out T item)
         /// <param name="millisecondsTimeout">The number of milliseconds to wait for the remove to complete.</param>
         /// <param name="cancellationToken">The cancellation token that can be used to cancel the operation.</param>
         /// <returns><c>true</c> if the remove completed within the specified timeout; otherwise, <c>false</c>.</returns>
-        public bool TryTake(out T item, int millisecondsTimeout, CancellationToken cancellationToken)
+        public async ValueTask<(bool Success, T Item)> TryTake(int millisecondsTimeout, CancellationToken cancellationToken)
         {
-            if(_collection.TryTake(out var p, millisecondsTimeout, cancellationToken))
+            if (await _collection.OutputAvailableAsync(cancellationToken))
```

Comment:
Missing waiting for ` millisecondsTimeout` in case not message yet!

Reply:
All the timeouts in my libraries are expressed as CancellationTokens https://github.com/StephenCleary/AsyncEx/issues/235#issuecomment-796238375